packages/gluestick/src/config/webpack/webpack.config.client.js

```diff
@@ -50,14 +52,35 @@ module.exports = (
     };
   });
   config.plugins.push(
-    new chunksPlugin(
-      deepClone(configuration),
-      { silent: settings.silent, chunk_info_filename: settings.chunk_info_filename },
-    ),
     // Make it so *.server.js files return null in client
     new webpack.NormalModuleReplacementPlugin(/\.server(\.js)?$/, path.join(__dirname, './mocks/serverFileMock.js')),
     progressHandler(logger, 'client'),
     new DuplicatePackageChecker(),
   );
+
+  // If vendor Dll bundle exists, use it otherwise fallback to CommonsChunkPlugin.
+  const vendorDllManifestPath: string = path.join(
+    process.cwd(), gluestickConfig.buildDllPath, manifestFilename,
+  );
+  if (fs.existsSync(vendorDllManifestPath)) {
+    logger.info('Vendor DLL bundle found, using DllReferencePlugin');
+    config.plugins.unshift(
+      new webpack.DllReferencePlugin({
+        context: configuration.context,
+        manifest: require(vendorDllManifestPath),
+      }),
+    );
+    config.plugins.push(new ChunksPlugin(deepClone(configuration), { appendChunkInfo: true }));
+  } else {
+    logger.info('Vendor DLL bundle not found, using CommonsChunkPlugin');
+    config.plugins.push(
+      new webpack.optimize.CommonsChunkPlugin({
+        name: 'vendor',
+        filename: `vendor${process.env.NODE_ENV === 'production' ? '-[hash]' : ''}.bundle.js`,
+      }),
```

Comment:
I think a good solution might be:
We do a very specific `vendor` common chunk using the instructions from the link above. We lock it down to matching _node_modules_. Then we need a way to exclude specific modules from the list. So in the `minChunks` function we match against node_modules and as long as it is not in the blacklist then it will get included.

This way the default behavior is the most common. Most of the time if multiple entry points share the same node_module then they should be extracted into vendor. However, when you come across an edge case where there is a couple of entrypoints that share something the rest of the apps should not, you can blacklist it. For example, a couple of apps might use numeraljs and moment.js which tend to be kind of big libraries, it would be nice to exclude those from the commonschunk if the majority of entry points do not need to use those.